QNX change from QNX's aports (core/fakeroot/fix-chown.patch).

Taken from qnx-ports/aports core/fakeroot -- the APKBUILD the QNX Everywhere 8.0.4r1
image's package is built from.

Upstream-Status: QNX-specific

--- fakeroot-1.38.1.orig/libfakeroot.c	2026-06-18 01:28:53.000000000 +0000
+++ fakeroot-1.38.1/libfakeroot.c	2026-06-18 01:33:30.000000000 +0000
@@ -862,8 +862,20 @@
     r=next_lchown(path,owner,group);
   else
     r=0;
+#ifdef __QNX__
+  // QNX bug that chown will return EACCES instead of EPERM on permission failure
+  if(r&&(errno==EACCES)){
+    // check if its really EACCESS
+    if (faccessat(AT_FDCWD, path, F_OK, 0) == EOK) {
+      r=0;
+    } else {
+      errno=EACCES;
+   }
+  }
+#else
   if(r&&(errno==EPERM))
     r=0;
+#endif

   return r;
 }
@@ -889,8 +901,21 @@
     r=next_lchown(path,owner,group);
   else
     r=0;
+#ifdef __QNX__
+  // QNX bug that lchown will return EACCES instead of EPERM on permission failure
+  if(r&&(errno==EACCES)) {
+    struct stat statbuf;
+    // check if its really EACCESS
+    if (fstatat(AT_FDCWD, path, &statbuf, AT_SYMLINK_NOFOLLOW) == EOK) {
+      r=0;
+    } else {
+      errno=EACCES;
+    }
+  }
+#else
   if(r&&(errno==EPERM))
     r=0;
+#endif

   return r;
 }
@@ -912,9 +937,15 @@
     r=next_fchown(fd,owner,group);
   else
     r=0;
-
+#ifdef __QNX__
+  // QNX bug that fchown will return EACCES instead of EPERM on permission failure
+  // we dont need a extra check since we already have a FD
+  if(r&&(errno==EACCES))
+      r=0;
+#else
   if(r&&(errno==EPERM))
     r=0;
+#endif

   return r;
 }
@@ -940,8 +971,21 @@
   else
     r=0;

+#ifdef __QNX__
+  // QNX bug that fchownat will return EACCES instead of EPERM on permission failure
+  if(r&&(errno==EACCES)) {
+    struct stat statbuf;
+    // check if its really EACCESS
+    if (fstatat(dir_fd, path, &statbuf, AT_SYMLINK_NOFOLLOW) == EOK) {
+      r=0;
+    } else {
+      errno=EACCES;
+    }
+  }
+#else
   if(r&&(errno==EPERM))
     r=0;
+#endif

   return r;
 }
